autonomy/analyse/abci/app_spec.py

```diff
@@ -338,6 +343,40 @@ def abci_to_dfa(cls, abci_app_cls: Any, label: str = "") -> "DFA":
         )
 
 
+event_pattern = re.compile(r"Event\.(\w+)", re.DOTALL)
+
+
+def _check_unreferenced_events(abci_app_cls: AbciApp) -> None:
+    """Checks that events defined in the AbciApp transition function are referenced in the source code of the coresponding round or its superclasses."""
+
+    error_strings = []
+    timeout_events = set([k.name for k in abci_app_cls.event_to_timeout.keys()])
```

Comment:
apart from list comprehension, you can use generator, set and dictionary comprehensions in python. It's more effective to use them directly, because now you generate the list first, then the set, which is wasteful.

this would make the linter happy
`timeout_events = {k.name for k in abci_app_cls.event_to_timeout}`

Reply:
Fixed.